packages/cli/tests/e2e/multienv/TestRemoteHappyPath.tests.ts

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import fs from "fs-extra";
+import path from "path";
+
+import {
+  AadValidator,
+  BotValidator,
+  FrontendValidator,
+  FunctionValidator,
+  SimpleAuthValidator,
+} from "../../commonlib";
+
+import {
+  cleanUp,
+  execAsync,
+  execAsyncWithRetry,
+  getSubscriptionId,
+  getTestFolder,
+  getUniqueAppName,
+  mockTeamsfxMultiEnvFeatureFlag,
+  setSimpleAuthSkuNameToB1Bicep,
+} from "../commonUtils";
+import AppStudioLogin from "../../../src/commonlib/appStudioLogin";
+import { deserializeDict, environmentManager } from "@microsoft/teamsfx-core";
+import {
+  err,
+  FxError,
+  Result,
+  ok,
+  ConfigFolderName,
+  PublishProfilesFolderName,
+  EnvProfileFileNameTemplate,
+  EnvNamePlaceholder,
+} from "@microsoft/teamsfx-api";
+
+// Load envProfile with userdata (not decrpted)
+async function loadContext(projectPath: string, env: string): Promise<Result<any, FxError>> {
+  const context = await fs.readJson(
+    path.join(
+      projectPath,
+      `.${ConfigFolderName}`,
+      PublishProfilesFolderName,
+      EnvProfileFileNameTemplate.replace(EnvNamePlaceholder, env)
+    )
+  );
+  const userdataContent = await fs.readFile(
+    path.join(projectPath, `.${ConfigFolderName}`, PublishProfilesFolderName, `${env}.userdata`),
+    "utf8"
+  );
+  const userdata = deserializeDict(userdataContent);
+
+  const regex = /\{\{([^{}]+)\}\}/;
+  for (const component in context) {
+    for (const key in context[component]) {
+      const matchResult = regex.exec(context[component][key]);
+      if (matchResult) {
+        const userdataKey = matchResult[1];
+        if (userdataKey in userdata) {
+          context[component][key] = userdata[key];
+        }
+      }
+    }
+  }
+  return ok(context);
+}
+
+describe("Create single tab/bot/function", function () {
```

Comment:
Seems the description doesn't match the behavior. You actually create a tab+bot+function.

Reply:
fixed